src/protocolhandler.ts

```diff
@@ -577,9 +578,17 @@ export class ProtocolHandler extends EventEmitter {
         let cmds: string[] = [];
         this.subscriptions.all().forEach((s) => {
             if (s.queue) {
-                cmds.push(`${SUB} ${s.subject} ${s.queue} ${s.sid} ${CR_LF}`);
+                cmds.push(`${SUB} ${s.subject} ${s.queue} ${s.sid}${CR_LF}`);
             } else {
-                cmds.push(`${SUB} ${s.subject} ${s.sid} ${CR_LF}`);
+                cmds.push(`${SUB} ${s.subject} ${s.sid}${CR_LF}`);
+            }
+            if (s.max) {
+                const max = s.max - s.received;
+                if (max > 0) {
+                    cmds.push(`${UNSUB} ${s.sid} ${max}${CR_LF}`);
+                } else {
+                    cmds.push(`${UNSUB} ${s.sid}${CR_LF}`);
```

Comment:
if this happens, get rid of s as well?

Reply:
The second UNSUB is really an impossibility because it would have meat that the dispatch loop saw a message finishing the subscription, yet we didn't process it. If it dispatched, then the sub would have been removed and would have never generated a resub.